spec/controllers/api/v1/project_roles_controller_spec.rb

```diff
@@ -94,6 +94,28 @@
       it_behaves_like "is creatable"
     end
 
+    context "when the user has an acl" do
+      let(:user) do
+        u = create(:user)
+        create(:access_control_list, user_group: u.identity_group, resource: project)
+        u.id.to_s
+      end
+      
+      before(:each) do
+        default_request user_id: authorized_user.id, scopes: scopes
+        post :create, create_params
+      end
+      
+      it 'should return 400' do
+        expect(response).to have_http_status(:bad_request)
+      end
+
+      it 'should have an error message' do
+        msg = json_response['errors'][0]['message']
+        expect(msg).to match(/Roles have already been set for this user or group/)
```

Comment:
there is a [helper method for these](https://github.com/zooniverse/Panoptes/blob/5f85c5558245cb75073d5501a58af1a3da468dda/spec/support/controller_helpers.rb#L6)